pkg/services/notary/core_test.go

```diff
@@ -591,23 +591,35 @@ func TestNotary(t *testing.T) {
 	// PostPersist: different NVBs
 	// check OnNewRequest with finalization error and different NVBs
 	setFinalizeWithError(true)
-	requests, requesters = checkCompleteStandardRequest(t, 5, false, 1, 2, 3, 4, 5)
+	// Introduce some slippage between first and second fallback NVBs in order to avoid possible race caused by early
+	// first fallback transaction acceptance. The rest of fallbacks follow X+3 NVB pattern for testing code shortness.
+	requests, requesters = checkCompleteStandardRequest(t, 5, false, 1, 6, 9, 12, 15)
 	checkFallbackTxs(t, requests, false)
 	// generate blocks to reach the most earlier fallback's NVB
-	e.GenerateNewBlocks(t, int(nvbDiffFallback))
+	e.GenerateNewBlocks(t, int(nvbDiffFallback+1))
 	require.NoError(t, err)
 	// check PostPersist for valid fallbacks without finalisation error
-	// Add block before allowing tx to finalize to exclude race condition when
-	// main transaction is finalized between `finalizeWithError` restore and adding new block.
+	// Add a block before allowing tx to be finalized without error to exclude race condition when
+	// main transaction is finalized between `finalizeWithError` disabling and new block addition.
 	e.AddNewBlock(t)
-	setFinalizeWithError(false)
-	for i := range requests[1:] {
+	mtx.RLock()
+	start := len(completedTxes)
+	finalizeWithError = false
+	mtx.RUnlock()
+
+	for i := range requests {
+		e.AddNewBlock(t)
 		e.AddNewBlock(t)
+		e.AddNewBlock(t)
+		require.Eventually(t, func() bool {
```

Comment:
And this can be avoided if you're to add one more block.

Reply:
Fixed.